lib/grape_entity/entity.rb

```diff
@@ -650,7 +650,14 @@ def options_for_using(attribute, options)
       using_options = options.dup
       using_options.delete(:collection)
       using_options[:root] = nil
-      using_options[:only] = only_fields(using_options)[attribute]
+      attribute_only_option = only_fields(using_options)[attribute]
+
+      if attribute_only_option.is_a?(Array)
+        using_options[:only] = attribute_only_option
+      else
+        using_options.delete(:only)
+      end
+
```

Comment:
You should sink the checking for `Array` inside `only_fields`, this way you don't need to check anything here. If that function returns `nil` you'll be assigning `using_options[:only] = nil` which is what we want.